CPP_STYLE_GUIDE.md

```diff
@@ -0,0 +1,138 @@
+# C++ style guide
+
+Unfortunately, the C++ linter (based on
+[Google’s `cpplint`](https://github.com/google/styleguide)), which can be run
+explicitly via `make lint-cpp`, does not currently catch a lot of rules that are
+specific to the Node.js C++ code base. This document explains the most common of
+these rules:
+
+## Left-leaning (C++ style) asterisks for pointer declarations
+
+`char* buffer;` instead of `char *buffer;`
+
+## 2 spaces of indentation for blocks or bodies of conditionals
```

Comment:
Also for loops (unless conditionals includes loops).

    while (something)
      ChangeSomething();

Now that I am thinking about it, I am not sure we use loops without parens...

Reply:
Yeah, conditionals includes conditional loops. :) Anyway, I don’t think the current phrasing leaves anybody thinking that we use indentation other than 2 spaces.